File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/annotations/Permit.java
package net.crystalapps.permission.runtime.library.annotations;

import androidx.annotation.StringRes;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */

@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.METHOD)
public @interface Permit {

    String value();
    @StringRes int whenDenied() default -1;
    @StringRes int whenPermanentDenied() default -1;
    boolean checkPermanentDenied() default true;
    boolean required() default true;
}

File: mint-runtime-permission/src/main/java/net/crystalapps/permission/runtime/library/annotations/Permit.kt
package net.crystalapps.permission.runtime.library.annotations

import androidx.annotation.StringRes
import java.lang.annotation.RetentionPolicy
import kotlin.annotation.Retention

/**
 * Created by Syed Owais Ali on 10/13/2018.
 */

@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class Permit(
        val value: String,
        @StringRes val whenDenied: Int = -1,
        @StringRes val whenPermanentDenied: Int = -1,
        val checkPermanentDenied: Boolean = true,
        val required: Boolean = true
)
